### PR TITLE
doc: app_dev: expand list of output files

### PR DESCRIPTION
**Pull request by greg-fer**
Added TF-M output files to the output file table.
NCSIDB-718.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/app_build_system.rst [hunk 1]
```diff
@@ -141,6 +147,12 @@ The following scenarios are possible:
 |                                 | (customizable by user, sample name by default).                                                 |                                                                          |
 |                                 | For example: :file:`127F-0141-01020003-light_switch.zigbee`.                                    |                                                                          |
 +---------------------------------+-------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------+
+| :file:`tfm_s.hex`               |                                                                                                 | DFU over Zigbee for both single-core and multi-core applications         |
```
Comment (joerchan) [inline doc/nrf/app_build_system.rst:150]: The "DFU over zigbee" line is a copy-paste error?

**Comment by umapraseeda** [comment]
Do we need a changelog entry? @balaji-nordic could you confirm if we can delete the table from ug_thingy91.rst?

**Review by balaji-nordic — CHANGES_REQUESTED** [review]
I prefer to keep the table in thingy user guide because it is more specific to the thingy use case. For example, many thingy users will have to use an 'external programmer' and 'nrf Connect programmer app' for programming. Since we have the right wording for that use cases in that table, and is very difficult to put those in the generic app_build_system.rst file, its better to keep that table.

**Comment by greg-fer** [comment]
> 

No problem, @balaji-nordic , as long as we link to the output build section from under the table. Restored the table.

**Review by balaji-nordic — APPROVED** [review]

**Review by divipillai — APPROVED** [review]
Just a minor comment.

**Comment by greg-fer** [comment]
@trond-snekvik , @joerchan , can you have a look? :)

**Review by umapraseeda — APPROVED** [review]

**Review by trond-snekvik — APPROVED** [review]

**Pull request merged**